src/core/jit/builder.rs

```diff
@@ -145,6 +166,30 @@ impl Builder {
                         continue;
                     }
 
+                    let mut directives = Vec::with_capacity(gql_field.directives.len());
+                    for directive in &gql_field.directives {
+                        let directive = &directive.node;
+                        if directive.name.node == "skip" || directive.name.node == "include" {
+                            continue;
+                        }
+                        let mut arguments = Vec::with_capacity(directive.arguments.len());
+                        for (name, value) in &directive.arguments {
+                            arguments.push((
+                                name.clone(),
+                                value.position_node(value.node.clone().into_const_with(
+                                    |name| {
+                                        self.get_variable_value(
```

Comment:
the `src/core/jit/input_resolver.rs` implements the same logic but only for `field.args`.

we need to merge this together somehow and remove duplication

Reply:
we can implement proper defaulting later, right now we should provide some abstractions to easier integrate this. Because of that I see the reason to use single entity to resolve variables in both args and directives